### DISCUSSION TITLE
What's the best way to have Handles act both as Source and Target?

### DISCUSSION THREAD

**Discussion by chris23lngr** [xyflow/xyflow#4345] [Q&A] [answered]
I am currently working on an app in which it is not important whether a handle is a target on or source. I use reactflow exclusively for the graphical representation of an entity-relationship model.

I want users to be able to create relationships between different entities by connecting the handles. Due to the user experience, it is poor if the user always has to search for the source handle of a node first. 

In your opinion, what is the best way to make a handle work as a source and target at the same time? I have already tried to layer several handles on top of each other, but this has led to many bugs.

**Comment by moklick** [comment, marked as answer]
Set all handle types to "source" and `connectionMode` to `ConnectionMode.Loose`.
